### DISCUSSION TITLE
修改页眉

### DISCUSSION THREAD

**Discussion by Skyliu1996** [tuna/thuthesis#768] [Q&A] [answered]
您好，我是外校的。我想请问下可不可以修改页眉，因为我们学校要求奇数页和偶数页的页眉分别固定为xx大学硕士研究生学位论文和论文标题。但是我尝试了下改不了，不知道有没有其他方法可以修改

**Comment by zepinglee** [comment, marked as answer]
https://github.com/tuna/thuthesis/blob/366b9413482584f3684346c0fc351d46cfbaaff9/thuthesis.dtx#L3175-L3216

一是注意 `\ifthu@degree@bachelor` 的条件；二是奇偶数页不同的话注意修改 `\@mkboth`、`\chaptermark` 和 `\sectionmark`，
比如标准文档类 https://github.com/latex3/latex2e/blob/4d5db49b784151c950f9415a57594439eebf7593/base/classes.dtx#L1688-L1701
```tex
    \def\chaptermark##1{%
      \markboth {\MakeUppercase{%
        \ifnum \c@secnumdepth >\m@ne
          \if@mainmatter
            \@chapapp\ \thechapter. \ %
          \fi
        \fi
        ##1}}{}}%
    \def\sectionmark##1{%
      \markright {\MakeUppercase{%
        \ifnum \c@secnumdepth >\z@
          \thesection. \ %
        \fi
        ##1}}}}
```
